"Compilation error: Error: 2 UNKNOWN: exit status 1" error message is unnecessarily cryptic

**Describe the bug**
Any program I run and attempt to compile I get this error:

> c:/users/anton/appdata/local/arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino7/bin/../lib/gcc/avr/7.3.0/../../../../avr/lib/avr5/crtatmega328p.o:(.init9+0x0): undefined reference to `main'
collect2.exe: error: ld returned 1 exit status
Compilation error: Error: 2 UNKNOWN: exit status 1

- I recently got the beta and no matter what I do I had this (yes I restarted my pc) and I even download the latest Git (2.30.1) 64-bit.

**To Reproduce**
Steps to reproduce the behavior:

1. Open any code
2. Click compile

**Expected behavior**
Expected for the code to compile.


**Desktop (please complete the following information):**
 - OS: Windows 10 Pro 64-bit
 - Version: 2.0.0-beta3


**Additional context**
This could be due to from when some notification occurred which I couldn't figure out what it was for. I do remember that it led to [this link](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc). However, I don't see this notification anymore?

Same issue here. A program that easily comiles and runs under the old IDE gets:

`collect2.exe: error: ld returned 1 exit status
Using library IRremote at version 3.0.3 in folder: C:\Users\ccbland\Documents\Arduino\libraries\IRremote 
Compilation error: Error: 2 UNKNOWN: exit status 1`

when compiled with V2. No other discernable complaints when using verbose mode on compile.

I did uninstall/reinstall IDE 2.0 to no avail.

Here's my program
```
#include "IRremote.h"

const int IR_RECEIVE_PIN = 11;
long keyPressed;


void setup()
{
    //setup serial monitor
    Serial.begin(115200);

    //initialize IR Receiver
    IrReceiver.begin(IR_RECEIVE_PIN, DISABLE_LED_FEEDBACK);
}

void loop()
{
    //check to see if a button was pressed
    if (IrReceiver.decode())
    {
        //a button press was pressed and decoded

        //get the code of the key that was pressed
        keyPressed = IrReceiver.decodedIRData.decodedRawData;

        //if that code is from the power button, tell us
        if (keyPressed==0xBA45FF00) Serial.println("Power Key Pressed");

        //wait for things to settle down
        delay(700);

        //resume receiving button presses
        IrReceiver.resume();
    }
}
```

--- a/arduino-ide-extension/src/browser/contributions/burn-bootloader.ts
+++ b/arduino-ide-extension/src/browser/contributions/burn-bootloader.ts
@@ -79,7 +79,13 @@ export class BurnBootloader extends SketchContribution {
         }
       );
     } catch (e) {
-      this.messageService.error(e.toString());
+      let errorMessage = "";
+      if (typeof e === "string") {
+        errorMessage = e;
+      } else {
+        errorMessage = e.toString();
+      }
+      this.messageService.error(errorMessage);
     } finally {
       if (this.serialConnection.isSerialOpen()) {
         await this.serialConnection.connect();

--- a/arduino-ide-extension/src/browser/contributions/upload-sketch.ts
+++ b/arduino-ide-extension/src/browser/contributions/upload-sketch.ts
@@ -277,7 +277,13 @@ export class UploadSketch extends SketchContribution {
         { timeout: 3000 }
       );
     } catch (e) {
-      this.messageService.error(e.toString());
+      let errorMessage = "";
+      if (typeof e === "string") {
+        errorMessage = e;
+      } else {
+        errorMessage = e.toString();
+      }
+      this.messageService.error(errorMessage);
     } finally {
       this.uploadInProgress = false;
       this.onDidChangeEmitter.fire();

--- a/arduino-ide-extension/src/browser/contributions/verify-sketch.ts
+++ b/arduino-ide-extension/src/browser/contributions/verify-sketch.ts
@@ -127,7 +127,13 @@ export class VerifySketch extends SketchContribution {
         { timeout: 3000 }
       );
     } catch (e) {
-      this.messageService.error(e.toString());
+      let errorMessage = "";
+      if (typeof e === "string") {
+        errorMessage = e;
+      } else {
+        errorMessage = e.toString();
+      }
+      this.messageService.error(errorMessage);
     } finally {
       this.verifyInProgress = false;
       this.onDidChangeEmitter.fire();

--- a/arduino-ide-extension/src/node/core-service-impl.ts
+++ b/arduino-ide-extension/src/node/core-service-impl.ts
@@ -23,6 +23,7 @@ import { NotificationServiceServer } from '../common/protocol';
 import { ArduinoCoreServiceClient } from './cli-protocol/cc/arduino/cli/commands/v1/commands_grpc_pb';
 import { firstToUpperCase, firstToLowerCase } from '../common/utils';
 import { Port } from './cli-protocol/cc/arduino/cli/commands/v1/port_pb';
+import { nls } from '@theia/core';
 
 @injectable()
 export class CoreServiceImpl extends CoreClientAware implements CoreService {
@@ -85,11 +86,16 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
         chunk: '\n--------------------------\nCompilation complete.\n',
       });
     } catch (e) {
+      const errorMessage = nls.localize(
+        'arduino/compile/error',
+        'Compilation error: {0}',
+        e.details
+      );
       this.responseService.appendToOutput({
-        chunk: `Compilation error: ${e.details}\n`,
+        chunk: `${errorMessage}}\n`,
         severity: 'error',
       });
-      throw e;
+      throw new Error(errorMessage);
     }
   }
 
@@ -180,11 +186,17 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
           ' complete.\n',
       });
     } catch (e) {
+      const errorMessage = nls.localize(
+        'arduino/upload/error',
+        '{0} error: {1}',
+        firstToUpperCase(task),
+        e.details,
+      );
       this.responseService.appendToOutput({
-        chunk: `${firstToUpperCase(task)} error: ${e.details}\n`,
+        chunk: `${errorMessage}\n`,
         severity: 'error',
       });
-      throw e;
+      throw new Error(errorMessage);
     } finally {
       this.uploading = false;
     }
@@ -227,11 +239,16 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
         result.on('end', () => resolve());
       });
     } catch (e) {
+      const errorMessage = nls.localize(
+        'arduino/burnBootloader/error',
+        'Error while burning the bootloader: {0}',
+        e.details,
+      );
       this.responseService.appendToOutput({
-        chunk: `Error while burning the bootloader: ${e.details}\n`,
+        chunk: `${errorMessage}\n`,
         severity: 'error',
       });
-      throw e;
+      throw new Error(errorMessage);
     }
   }
 

--- a/i18n/en.json
+++ b/i18n/en.json
@@ -288,6 +288,15 @@
         "electron": {
             "couldNotSave": "Could not save the sketch. Please copy your unsaved work into your favorite text editor, and restart the IDE.",
             "unsavedChanges": "Any unsaved changes will not be saved."
+        },
+        "compile": {
+            "error": "Compilation error: {0}"
+        },
+        "upload": {
+            "error": "{0} error: {1}"
+        },
+        "burnBootloader": {
+            "error": "Error while burning the bootloader: {0}"
         }
     },
     "theia": {

